src/test/ruby/test_asn1.rb

```diff
@@ -378,16 +410,13 @@ def test_decode
     assert_equal(65537, spkey.value[1].value)
 
     extensions = tbs_cert.value[7]
-    assert_equal(:CONTEXT_SPECIFIC, extensions.tag_class)
+    assert_equal(:UNIVERSAL, extensions.tag_class)
     assert_equal(3, extensions.tag)
-    assert_equal(1, extensions.value.size)
     assert_equal(OpenSSL::ASN1::Sequence, extensions.value[0].class)
-    assert_equal(3, extensions.value[0].value.size)
+    assert_equal(3, extensions.value[0].size)
```

Comment:
:warning: a lot of the `.value` wrapping changes (such as in this test) seem to me like they are regressions as this has and still (using OpenSSL 3.0) passes on MRI.